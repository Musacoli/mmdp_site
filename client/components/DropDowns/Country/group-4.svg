<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
    <g fill="none" fill-rule="evenodd">
        <circle cx="16" cy="16" r="16" fill="#84C8DD"/>
        <g fill="#FFF">
            <path fill-rule="nonzero" d="M22.49 8.556h-2.683v-.539c0-.892-.723-1.617-1.61-1.617h-4.294c-.887 0-1.61.725-1.61 1.617v.54H9.61c-.888 0-1.61.725-1.61 1.616a1.62 1.62 0 0 0 1.11 1.537l.957 11.607A1.622 1.622 0 0 0 11.67 24.8h8.758c.83 0 1.536-.651 1.605-1.483l.957-11.607a1.619 1.619 0 0 0-.501-3.154zm-9.123-.539c0-.297.24-.539.536-.539h4.294c.296 0 .536.242.536.54v.538h-5.366v-.539zm7.597 15.21a.54.54 0 0 1-.535.495h-8.758a.54.54 0 0 1-.535-.494l-.943-11.437h11.714l-.943 11.437zm1.526-12.515H9.61a.538.538 0 0 1 0-1.078h12.88a.538.538 0 0 1 0 1.078z"/>
            <path d="M13.877 21.931l-.511-8.64a.52.52 0 0 0-.542-.501.527.527 0 0 0-.479.568l.512 8.64a.522.522 0 0 0 .51.502.526.526 0 0 0 .51-.569zM16.178 12.789a.524.524 0 0 0-.511.535v8.64c0 .296.229.536.51.536.283 0 .512-.24.512-.536v-8.64a.524.524 0 0 0-.511-.535zM19.276 12.79a.52.52 0 0 0-.542.501l-.51 8.64a.527.527 0 0 0 .478.568.52.52 0 0 0 .541-.501l.512-8.64a.527.527 0 0 0-.479-.568z"/>
        </g>
    </g>
</svg>
