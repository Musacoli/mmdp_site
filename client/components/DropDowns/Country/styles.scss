
#more__options{
  // background-color: red;
  width: 132px;
  height: 29px;
  font-family: Muli;
  font-size: 16px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.81;
  letter-spacing: -0.3px;
  text-align: center;
  color:#3faaca;
  margin-right:30px;
}


#save__dropdown{
  width: 200px;
  font-family: Muli;
  font-size: 16px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.81;
  letter-spacing: -0.3px;
  text-align: center;
  color: #fff;
}

#delete__logo{
  margin-top: 25px;
}
