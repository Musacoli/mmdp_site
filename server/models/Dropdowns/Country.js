import keystone from 'keystone';

const Country = new keystone.List('Country');

Country.add({
  countryName: { type: String },
  description: { type: String },
});

Country.defaultColumns = 'countryName';

Country.register();

export default Country;
