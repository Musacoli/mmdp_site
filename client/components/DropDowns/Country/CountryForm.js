import React from 'react';
import { Grid, Button, Input } from 'semantic-ui-react';
import './styles.scss';
import deletelogo from './group-4.svg';

export const EditDropDownForm = (props) => (
  <>
    <Grid columns="equal">
      <Grid.Column width={7}>
        <label htmlFor="country__name">Country Name</label>
        <input
          type="text"
          id="country__name"
          sharedid={props.itemId}
          className="ui input"
          placeholder="country name"
          name="countryName"
          onChange={props.onchangeHandler}
          defaultValue={props.countryName || ''}
        />
      </Grid.Column>
      <Grid.Column width={7}>
        <label htmlFor="country__description">Description </label>
        <input
          type="text"
          id="country__description"
          sharedid={props.itemId}
          className="ui input"
          placeholder="Description"
          name="countryDescription"
          onChange={props.onchangeHandler}
          defaultValue={props.countryDescription || ''}
        />
      </Grid.Column>
      <Grid.Column>
        <img
          src={deletelogo}
          alt="logo"
          title="delete"
          id="delete__logo"
          sharedid={props.itemId}
          onClick={props.delete}
        />
      </Grid.Column>
    </Grid>
  </>
);

export const ButtonHolder = ({ click }) => (
  <Grid stackable columns={1}>
    <Grid.Column>
      <Grid.Row>
        <a href="#" onClick={click} id="more__options">
          Add more options
        </a>
        <Button type="submit" id="save__dropdown">
          Save dropdown
        </Button>
      </Grid.Row>
    </Grid.Column>
  </Grid>
);
