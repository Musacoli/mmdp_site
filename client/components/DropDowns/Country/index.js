import React from 'react';
import { Form, Grid } from 'semantic-ui-react';
import CountryForm from '../../common/Form/DropdownForm';
import DropdownActions from '../../common/Button/DropdownActions';

const Country = (props) => {
  const {
    dropdowns,
    addTempState,
    handleSubmit,
    twoFieldsClass,
    loading,
  } = props;
  return (
    <Grid.Row className={loading ? 'dropdowns ui form loading' : 'dropdowns'}>
      <Form>
        {dropdowns.map((item, index) => {
          const inputs = [
            {
              type: 'text',
              name: 'countryName',
              label: 'Country name',
              className: 'animated fadeIn',
              fluid: true,
              placeholder: 'Enter country name',
              value: item.countryName || '',
            },
            {
              type: 'text',
              name: 'description',
              label: 'Description',
              className: 'animated fadeIn',
              fluid: true,
              placeholder: 'description',
              value: item.description || '',
            },
          ];
          return (
            <CountryForm
              // eslint-disable-next-line
              key={index}
              item={item}
              inputs={inputs || []}
              editAState={props.editAStatus}
              deleteAState={props.deleteAStatus}
              className={twoFieldsClass}
              header="Delete Country"
              label="Confirm Delete Country"
            />
          );
        })}
        <DropdownActions
          addDropdown={addTempState}
          handleSubmit={handleSubmit}
        />
      </Form>
    </Grid.Row>
  );
};

export default Country;
