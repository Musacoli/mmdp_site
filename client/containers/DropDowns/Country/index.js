import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as countryActions from '../../../store/actions/dropdowns/country';
import CountryForm from '../../../components/DropDowns/Country';
import { filterDropdowns, hasId, hasNoId } from '../../../utils/dropdowns';

export class CountryDropDown extends Component {
  static propTypes = {
    fetchCountry: PropTypes.func,
    addCountry: PropTypes.func,
    deleteCountry: PropTypes.func,
  };

  state = {
    dropdowns: [],
    refresh: true,
  };

  componentDidMount() {
    const { fetchCountry } = this.props;
    fetchCountry();
  }

  componentDidUpdate() {
    const { loading, countries } = this.props;
    const { dropdowns, refresh } = this.state;
    const filtered = filterDropdowns(dropdowns, hasId);

    if (
      countries.length > 0 &&
      !loading &&
      refresh &&
      filtered.length !== countries.length
    ) {
      // eslint-disable-next-line
      this.setState({ dropdowns: countries, refresh: false });
    } else if (countries.length === 0 && !loading && refresh) {
      // eslint-disable-next-line
      this.setState({ dropdowns: countries, refresh: false });
    }
  }

  addTempState = () => {
    const { dropdowns } = this.state;
    const countries = dropdowns.slice();
    countries.push({
      countryName: '',
      description: '',
      id: dropdowns.length + 1,
    });
    this.setState({ dropdowns: countries });
  };

  editAStatus = (dropdown, deleteErrors = true) => {
    const { dropdowns } = this.state;
    const countries = dropdowns.slice();

    const data = { ...dropdown };
    const key = dropdown._id ? '_id' : 'id';
    const stateIndex = countries.findIndex((item) => item[key] === data[key]);
    if (deleteErrors) {
      delete data.errors;
    }

    countries[stateIndex] = data;
    this.setState({ dropdowns: countries });
  };

  deleteAStatus = (item) => {
    const { deleteCountry } = this.props;
    const { dropdowns } = this.state;
    const countries = dropdowns.slice();
    if (item._id) {
      deleteCountry({ id: item._id });
      this.setState({ refresh: true });
      return false;
    }
    const index = countries.indexOf(item);
    if (index > -1) {
      countries.splice(index, 1);
    }
    this.setState({ dropdowns: countries });
  };

  handleSubmit = () => {
    const { dropdowns } = this.state;
    const { addCountry } = this.props;
    const countries = [];
    const errors = [];

    // validate field required
    dropdowns.map((item) => {
      const data = item;
      if (data.countryName === '' || data.countryName === undefined) {
        data.errors = data.errors ? data.errors : {};
        data.errors.countryName = 'Please enter a Country Name ';
        errors.push(data.errors.countryName);
      }

      return countries.push(data);
    });
    // validate stateName duplicates
    const editData = filterDropdowns(countries, hasId);
    const addData = filterDropdowns(countries, hasNoId);

    editData.map((item) => {
      function condition(countryData) {
        return countryData.countryName === item.countryName;
      }
      const duplicates = addData.filter(condition);
      // console.warn(duplicates)
      return duplicates.map((value) => {
        // eslint-disable-next-line
        value.errors = value.errors ? value.errors : {};
        // eslint-disable-next-line
        value.errors.countryName = `${value.countryName} country exists`;
        errors.push(value);
        return this.editAStatus(value, false);
      });
    });
    this.setState({ dropdowns: countries });
    if (errors.length < 1) {
      if (editData.length > 0) {
        addCountry({ data: { data: editData }, new: false });
      }
      if (addData.length > 0) {
        addCountry({ data: { data: addData }, new: true });
        this.setState({ refresh: true });
      }
    }
  };

  render() {
    const { dropdowns } = this.state;
    return (
      <CountryForm
        dropdowns={dropdowns}
        {...this.props}
        addTempState={this.addTempState}
        editAStatus={this.editAStatus}
        handleSubmit={this.handleSubmit}
        deleteAStatus={this.deleteAStatus}
        twoFieldsClass="two-fields"
      />
    );
  }
}

const mapStateToProps = (state) => ({
  countries: state.country.data,
  loading: state.country.loading,
});

const mapDispatchToProps = {
  fetchCountry: countryActions.fetchCountry,
  addCountry: countryActions.addCountry,
  deleteCountry: countryActions.deleteCountry,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CountryDropDown);
