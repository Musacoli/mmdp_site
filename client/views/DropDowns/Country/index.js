import React from 'react';
import Templates from '../../Templates';
import Country from '../../../containers/DropDowns/Country';

const CountryDropDown = ({ ...props }) => (
  <Templates {...props} title="Edit Country Dropdown">
    <Country {...props} />
  </Templates>
);

export default CountryDropDown;
