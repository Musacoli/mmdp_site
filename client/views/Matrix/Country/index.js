import React from 'react';
import Templates from '../../Templates';
import Country from '../../../containers/Matrix/CountryMap';
import '../../../assets/styles/components/map_listing.scss';

const CountryMatrixView = ({ ...props }) => (
  <Templates {...props} title="National Level">
    <Country {...props} />
  </Templates>
);

export default CountryMatrixView;
