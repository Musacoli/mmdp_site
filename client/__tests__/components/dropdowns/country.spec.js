import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Country from '../../../components/DropDowns/country';

describe('<countryComponent />', () => {
  const props = {
    dropdowns: [
      {
        countryId: '5c878ecd6d5a6b1184b93519',
        countryName: 'Ug',
        description: 'yoooooo',
        __v: 0,
        _id: '5c90de765a04a53d87040c5e',
      },
      {
        countryId: '5c878ecd6d5a6b1184b93519',
        countryName: 'Ke',
        description: 'East africa',
        __v: 0,
        id: '5c90de765a04a53d87040c5e',
      },
    ],
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    addNewDropDown: jest.fn(),
    handleDelete: jest.fn(),
    loading: false,
  };
  const wrapper = mount(<Country {...props} />);
  it('should render country component without crashing', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
