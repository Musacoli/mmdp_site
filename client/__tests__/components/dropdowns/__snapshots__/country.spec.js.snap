// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<countryComponent /> should render country component without crashing 1`] = `
<Country
  addNewDropDown={[MockFunction]}
  dropdowns={
    Array [
      Object {
        "__v": 0,
        "_id": "5c90de765a04a53d87040c5e",
        "countryId": "5c878ecd6d5a6b1184b93519",
        "countryName": "Ug",
        "description": "yoooooo",
      },
      Object {
        "__v": 0,
        "countryId": "5c878ecd6d5a6b1184b93519",
        "countryName": "Ke",
        "description": "East africa",
        "id": "5c90de765a04a53d87040c5e",
      },
    ]
  }
  handleChange={[MockFunction]}
  handleDelete={[MockFunction]}
  handleSubmit={[MockFunction]}
  loading={false}
>
  <GridRow
    className="dropdowns"
  >
    <div
      className="row dropdowns"
    >
      <Form
        as="form"
      >
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <DropdownForm
            header="Delete Country"
            inputs={
              Array [
                Object {
                  "className": "animated fadeIn",
                  "fluid": true,
                  "label": "Country name",
                  "name": "countryName",
                  "placeholder": "Enter country name",
                  "type": "text",
                  "value": "Ug",
                },
                Object {
                  "className": "animated fadeIn",
                  "fluid": true,
                  "label": "Description",
                  "name": "description",
                  "placeholder": "description",
                  "type": "text",
                  "value": "yoooooo",
                },
              ]
            }
            item={
              Object {
                "__v": 0,
                "_id": "5c90de765a04a53d87040c5e",
                "countryId": "5c878ecd6d5a6b1184b93519",
                "countryName": "Ug",
                "description": "yoooooo",
              }
            }
            key="0"
            label="Confirm Delete Country"
          >
            <FormGroup
              widths="equal"
            >
              <div
                className="equal width fields"
              >
                <Input
                  attributes={
                    Object {
                      "className": "animated fadeIn",
                      "fluid": true,
                      "label": "Country name",
                      "name": "countryName",
                      "placeholder": "Enter country name",
                      "type": "text",
                      "value": "Ug",
                    }
                  }
                  dropdownItem={
                    Object {
                      "__v": 0,
                      "_id": "5c90de765a04a53d87040c5e",
                      "countryId": "5c878ecd6d5a6b1184b93519",
                      "countryName": "Ug",
                      "description": "yoooooo",
                    }
                  }
                  handleChange={[Function]}
                >
                  <FormField>
                    <div
                      className="field"
                    >
                      <FormInput
                        as={[Function]}
                        className="animated fadeIn"
                        control={[Function]}
                        fluid={true}
                        label="Country name"
                        name="countryName"
                        onChange={[Function]}
                        placeholder="Enter country name"
                        type="text"
                        value="Ug"
                      >
                        <FormField
                          className="animated fadeIn"
                          control={[Function]}
                          fluid={true}
                          label="Country name"
                          name="countryName"
                          onChange={[Function]}
                          placeholder="Enter country name"
                          type="text"
                          value="Ug"
                        >
                          <div
                            className="field animated fadeIn"
                          >
                            <label>
                              Country name
                            </label>
                            <Input
                              fluid={true}
                              name="countryName"
                              onChange={[Function]}
                              placeholder="Enter country name"
                              type="text"
                              value="Ug"
                            >
                              <div
                                className="ui fluid input"
                              >
                                <input
                                  name="countryName"
                                  onChange={[Function]}
                                  placeholder="Enter country name"
                                  type="text"
                                  value="Ug"
                                />
                              </div>
                            </Input>
                          </div>
                        </FormField>
                      </FormInput>
                    </div>
                  </FormField>
                </Input>
                 
                <Input
                  attributes={
                    Object {
                      "className": "animated fadeIn",
                      "fluid": true,
                      "label": "Description",
                      "name": "description",
                      "placeholder": "description",
                      "type": "text",
                      "value": "yoooooo",
                    }
                  }
                  dropdownItem={
                    Object {
                      "__v": 0,
                      "_id": "5c90de765a04a53d87040c5e",
                      "countryId": "5c878ecd6d5a6b1184b93519",
                      "countryName": "Ug",
                      "description": "yoooooo",
                    }
                  }
                  handleChange={[Function]}
                >
                  <FormField>
                    <div
                      className="field"
                    >
                      <FormInput
                        as={[Function]}
                        className="animated fadeIn"
                        control={[Function]}
                        fluid={true}
                        label="Description"
                        name="description"
                        onChange={[Function]}
                        placeholder="description"
                        type="text"
                        value="yoooooo"
                      >
                        <FormField
                          className="animated fadeIn"
                          control={[Function]}
                          fluid={true}
                          label="Description"
                          name="description"
                          onChange={[Function]}
                          placeholder="description"
                          type="text"
                          value="yoooooo"
                        >
                          <div
                            className="field animated fadeIn"
                          >
                            <label>
                              Description
                            </label>
                            <Input
                              fluid={true}
                              name="description"
                              onChange={[Function]}
                              placeholder="description"
                              type="text"
                              value="yoooooo"
                            >
                              <div
                                className="ui fluid input"
                              >
                                <input
                                  name="description"
                                  onChange={[Function]}
                                  placeholder="description"
                                  type="text"
                                  value="yoooooo"
                                />
                              </div>
                            </Input>
                          </div>
                        </FormField>
                      </FormInput>
                    </div>
                  </FormField>
                </Input>
                 
                <div
                  className="dropdowns__actions undefined __two_fields"
                >
                  <ActionModal
                    className=""
                    confirmDeleteGroup={[Function]}
                    content="Confirm Delete Country"
                    group={
                      Object {
                        "__v": 0,
                        "_id": "5c90de765a04a53d87040c5e",
                        "countryId": "5c878ecd6d5a6b1184b93519",
                        "countryName": "Ug",
                        "description": "yoooooo",
                      }
                    }
                    header="Delete Country"
                    triggerText={
                      <Icon
                        as="i"
                        name="trash alternate outline"
                        size="large"
                      />
                    }
                  >
                    <Modal
                      centered={true}
                      className="show"
                      closeOnDimmerClick={true}
                      closeOnDocumentClick={false}
                      dimmer={true}
                      eventPool="Modal"
                      open={false}
                      size="mini"
                      trigger={
                        <span
                          className=""
                          onClick={[Function]}
                          role="presentation"
                        >
                          <Icon
                            as="i"
                            name="trash alternate outline"
                            size="large"
                          />
                        </span>
                      }
                    >
                      <Portal
                        closeOnDocumentClick={false}
                        closeOnEscape={true}
                        eventPool="Modal"
                        mountNode={<body />}
                        onClose={[Function]}
                        onMount={[Function]}
                        onOpen={[Function]}
                        onUnmount={[Function]}
                        open={false}
                        openOnTriggerClick={true}
                        trigger={
                          <span
                            className=""
                            onClick={[Function]}
                            role="presentation"
                          >
                            <Icon
                              as="i"
                              name="trash alternate outline"
                              size="large"
                            />
                          </span>
                        }
                      >
                        <Ref
                          innerRef={[Function]}
                        >
                          <span
                            className=""
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            role="presentation"
                          >
                            <Icon
                              as="i"
                              name="trash alternate outline"
                              size="large"
                            >
                              <i
                                aria-hidden="true"
                                className="trash alternate outline large icon"
                              />
                            </Icon>
                          </span>
                        </Ref>
                      </Portal>
                    </Modal>
                  </ActionModal>
                </div>
              </div>
            </FormGroup>
          </DropdownForm>
          <DropdownForm
            header="Delete Country"
            inputs={
              Array [
                Object {
                  "className": "animated fadeIn",
                  "fluid": true,
                  "label": "Country name",
                  "name": "countryName",
                  "placeholder": "Enter country name",
                  "type": "text",
                  "value": "Ke",
                },
                Object {
                  "className": "animated fadeIn",
                  "fluid": true,
                  "label": "Description",
                  "name": "description",
                  "placeholder": "description",
                  "type": "text",
                  "value": "East africa",
                },
              ]
            }
            item={
              Object {
                "__v": 0,
                "countryId": "5c878ecd6d5a6b1184b93519",
                "countryName": "Ke",
                "description": "East africa",
                "id": "5c90de765a04a53d87040c5e",
              }
            }
            key="1"
            label="Confirm Delete Country"
          >
            <FormGroup
              widths="equal"
            >
              <div
                className="equal width fields"
              >
                <Input
                  attributes={
                    Object {
                      "className": "animated fadeIn",
                      "fluid": true,
                      "label": "Country name",
                      "name": "countryName",
                      "placeholder": "Enter country name",
                      "type": "text",
                      "value": "Ke",
                    }
                  }
                  dropdownItem={
                    Object {
                      "__v": 0,
                      "countryId": "5c878ecd6d5a6b1184b93519",
                      "countryName": "Ke",
                      "description": "East africa",
                      "id": "5c90de765a04a53d87040c5e",
                    }
                  }
                  handleChange={[Function]}
                >
                  <FormField>
                    <div
                      className="field"
                    >
                      <FormInput
                        as={[Function]}
                        className="animated fadeIn"
                        control={[Function]}
                        fluid={true}
                        label="Country name"
                        name="countryName"
                        onChange={[Function]}
                        placeholder="Enter country name"
                        type="text"
                        value="Ke"
                      >
                        <FormField
                          className="animated fadeIn"
                          control={[Function]}
                          fluid={true}
                          label="Country name"
                          name="countryName"
                          onChange={[Function]}
                          placeholder="Enter country name"
                          type="text"
                          value="Ke"
                        >
                          <div
                            className="field animated fadeIn"
                          >
                            <label>
                              Country name
                            </label>
                            <Input
                              fluid={true}
                              name="countryName"
                              onChange={[Function]}
                              placeholder="Enter country name"
                              type="text"
                              value="Ke"
                            >
                              <div
                                className="ui fluid input"
                              >
                                <input
                                  name="countryName"
                                  onChange={[Function]}
                                  placeholder="Enter country name"
                                  type="text"
                                  value="Ke"
                                />
                              </div>
                            </Input>
                          </div>
                        </FormField>
                      </FormInput>
                    </div>
                  </FormField>
                </Input>
                 
                <Input
                  attributes={
                    Object {
                      "className": "animated fadeIn",
                      "fluid": true,
                      "label": "Description",
                      "name": "description",
                      "placeholder": "description",
                      "type": "text",
                      "value": "East africa",
                    }
                  }
                  dropdownItem={
                    Object {
                      "__v": 0,
                      "countryId": "5c878ecd6d5a6b1184b93519",
                      "countryName": "Ke",
                      "description": "East africa",
                      "id": "5c90de765a04a53d87040c5e",
                    }
                  }
                  handleChange={[Function]}
                >
                  <FormField>
                    <div
                      className="field"
                    >
                      <FormInput
                        as={[Function]}
                        className="animated fadeIn"
                        control={[Function]}
                        fluid={true}
                        label="Description"
                        name="description"
                        onChange={[Function]}
                        placeholder="description"
                        type="text"
                        value="East africa"
                      >
                        <FormField
                          className="animated fadeIn"
                          control={[Function]}
                          fluid={true}
                          label="Description"
                          name="description"
                          onChange={[Function]}
                          placeholder="description"
                          type="text"
                          value="East africa"
                        >
                          <div
                            className="field animated fadeIn"
                          >
                            <label>
                              Description
                            </label>
                            <Input
                              fluid={true}
                              name="description"
                              onChange={[Function]}
                              placeholder="description"
                              type="text"
                              value="East africa"
                            >
                              <div
                                className="ui fluid input"
                              >
                                <input
                                  name="description"
                                  onChange={[Function]}
                                  placeholder="description"
                                  type="text"
                                  value="East africa"
                                />
                              </div>
                            </Input>
                          </div>
                        </FormField>
                      </FormInput>
                    </div>
                  </FormField>
                </Input>
                 
                <div
                  className="dropdowns__actions undefined __two_fields"
                >
                  <ActionModal
                    className=""
                    confirmDeleteGroup={[Function]}
                    content="Confirm Delete Country"
                    group={
                      Object {
                        "__v": 0,
                        "countryId": "5c878ecd6d5a6b1184b93519",
                        "countryName": "Ke",
                        "description": "East africa",
                        "id": "5c90de765a04a53d87040c5e",
                      }
                    }
                    header="Delete Country"
                    triggerText={
                      <Icon
                        as="i"
                        name="trash alternate outline"
                        size="large"
                      />
                    }
                  >
                    <Modal
                      centered={true}
                      className="show"
                      closeOnDimmerClick={true}
                      closeOnDocumentClick={false}
                      dimmer={true}
                      eventPool="Modal"
                      open={false}
                      size="mini"
                      trigger={
                        <span
                          className=""
                          onClick={[Function]}
                          role="presentation"
                        >
                          <Icon
                            as="i"
                            name="trash alternate outline"
                            size="large"
                          />
                        </span>
                      }
                    >
                      <Portal
                        closeOnDocumentClick={false}
                        closeOnEscape={true}
                        eventPool="Modal"
                        mountNode={<body />}
                        onClose={[Function]}
                        onMount={[Function]}
                        onOpen={[Function]}
                        onUnmount={[Function]}
                        open={false}
                        openOnTriggerClick={true}
                        trigger={
                          <span
                            className=""
                            onClick={[Function]}
                            role="presentation"
                          >
                            <Icon
                              as="i"
                              name="trash alternate outline"
                              size="large"
                            />
                          </span>
                        }
                      >
                        <Ref
                          innerRef={[Function]}
                        >
                          <span
                            className=""
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            role="presentation"
                          >
                            <Icon
                              as="i"
                              name="trash alternate outline"
                              size="large"
                            >
                              <i
                                aria-hidden="true"
                                className="trash alternate outline large icon"
                              />
                            </Icon>
                          </span>
                        </Ref>
                      </Portal>
                    </Modal>
                  </ActionModal>
                </div>
              </div>
            </FormGroup>
          </DropdownForm>
          <DropdownActions
            handleSubmit={[MockFunction]}
          >
            <FormGroup>
              <div
                className="fields"
              >
                <FormButton
                  as={[Function]}
                  className="dropdowns__add button-more"
                  control={[Function]}
                  onClick={[Function]}
                >
                  <FormField
                    className="dropdowns__add button-more"
                    control={[Function]}
                    onClick={[Function]}
                  >
                    <div
                      className="field dropdowns__add button-more"
                    >
                      <Button
                        as="button"
                        onClick={[Function]}
                      >
                        <button
                          className="ui button"
                          onClick={[Function]}
                        >
                          Add more options
                        </button>
                      </Button>
                    </div>
                  </FormField>
                </FormButton>
                <FormButton
                  as={[Function]}
                  className="dropdowns__save bg-transparent"
                  control={[Function]}
                  onClick={[MockFunction]}
                >
                  <FormField
                    className="dropdowns__save bg-transparent"
                    control={[Function]}
                    onClick={[MockFunction]}
                  >
                    <div
                      className="field dropdowns__save bg-transparent"
                    >
                      <Button
                        as="button"
                        onClick={[MockFunction]}
                      >
                        <button
                          className="ui button"
                          onClick={[Function]}
                        >
                          Save dropdown
                        </button>
                      </Button>
                    </div>
                  </FormField>
                </FormButton>
              </div>
            </FormGroup>
          </DropdownActions>
        </form>
      </Form>
    </div>
  </GridRow>
</Country>
`;
